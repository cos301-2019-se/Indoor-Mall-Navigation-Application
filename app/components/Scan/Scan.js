'use strict';
import React from 'react';
import { AppRegistry, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { RNCamera } from 'react-native-camera';

export default class Scan extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <RNCamera
          ref={ref => {
            this.camera = ref;
          }}
          style={styles.preview}
          type={RNCamera.Constants.Type.back}
          flashMode={RNCamera.Constants.FlashMode.on}
          androidCameraPermissionOptions={{
            title: 'Permission to use camera',
            message: 'We need your permission to use your camera',
            buttonPositive: 'Ok',
            buttonNegative: 'Cancel',
          }}
          onBarCodeRead={({ data, rawData, type, bounds }) => {
            alert(data);
          }}
        />
      </View>
    );
  }

  
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: 'black',
  },
  preview: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  capture: {
    flex: 0,
    backgroundColor: '#fff',
    borderRadius: 5,
    padding: 15,
    paddingHorizontal: 20,
    alignSelf: 'center',
    margin: 20,
  },
});