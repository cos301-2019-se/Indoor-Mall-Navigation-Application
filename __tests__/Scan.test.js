import React from 'react';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });


import Scan from '../app/components/Scan';


describe('<Scan />', () => {
	it('Scan renders without crashing', () => {
		const rendered = shallow(<Scan />);
		expect(rendered).toBeTruthy();
	});
	it('renders correctly', () => {
		const tree = shallow(<Scan />);
		expect(tree).toMatchSnapshot();
	});


});
