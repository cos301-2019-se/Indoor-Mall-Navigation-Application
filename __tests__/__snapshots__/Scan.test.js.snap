// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Scan /> renders correctly 1`] = `ShallowWrapper {}`;
